### DISCUSSION TITLE
diff bw search and view rule?

### DISCUSSION THREAD

**Discussion by Qa5imm** [pocketbase/pocketbase#3701] [Q&A] [answered]
what's the distinction between search and view api rule, I somewhat understand the view rule but not sure where/how should one use search rule

**Comment by ganigeorgiev** [comment]
The View API rule is used usually when querying the record by its id (eg. `GET /api/collections/example/records/ID` or `pb.collection("example").getOne(ID)`).

The List/Search API rule is used usually when performing list/search operations across all records in the collection (eg. `GET /api/collections/example/records?filter=...` or `pb.collection("example").getList(1, 30, { filter: "..." })` )

**Reply by Qa5imm** [reply]
will it work even if we don't apply the filter, I mean just on getList(1,30)

**Reply by ganigeorgiev** [reply, marked as answer]
Yes, the `filter` parameter was just an example (see [List/Search records](https://pocketbase.io/docs/api-records/#listsearch-records)).
